src/targets/gpu/kernels/include/migraphx/kernels/compute_attention_probabilities.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#ifndef MIGRAPHX_GUARD_KERNELS_COMPUTE_ATTENTION_PROBABILITIES_HPP
+#define MIGRAPHX_GUARD_KERNELS_COMPUTE_ATTENTION_PROBABILITIES_HPP
+
+#include <migraphx/kernels/group_query_attention.hpp>
+#include <migraphx/kernels/index.hpp>
+#include <migraphx/kernels/tensor_view.hpp>
+
+namespace migraphx {
+
+template <class AttnProbs,
+          class Query,
+          class SeqLensK,
+          class PresentKey,
+          class Params>
+__device__ void
+calculate_attention_probs(AttnProbs attention_probs, // output buffer with size BxNxSxT
+                          Query query,               // Q data. Its size is BxNxSxH
+                          SeqLensK seqlens_k,        // past sequence lengths tensor
+                          PresentKey present_key,    // present key only
+                          Params params,
+                          index_int idx)
+{
+    const int batch_size                        = params.batch_size;
+    const int sequence_length                   = params.sequence_length;
+    const int head_size                         = params.head_size;
+    const size_t present_buffer_sequence_length = params.seqlen_present_kv_cache;
+    const int num_heads                         = params.num_heads;
+    const int kv_num_heads                      = params.kv_num_heads;
+    const int packed_batch_stride = (num_heads + 2 * kv_num_heads) * sequence_length * head_size;
+    const int kv_num_heads_factor = num_heads / kv_num_heads;
+    const size_t q_input_chunk_length = static_cast<size_t>(sequence_length) * head_size; // S x H
```

Comment:
Why static_cast to `size_t` when you can just declare the original variable as `size_t`?